Khan/wonder-stuff
# Flow to TypeScript 2: Generate .d.ts and .js.flow files

## Summary:
The purpose of generating .js.flow files is to allow us to use types from wonder-stuff-core in Flow projects after converting wonder-stuff-core to TypeScript.

The .d.ts file was generated using:

    dts-bundle-generator -o wonder-stuff-core.d.ts ./packages/wonder-stuff-core/src/index.ts

The .js.flow file was generated using:

    flowgen -o wonder-stuff-core.js.flow wonder-stuff-core.d.ts

I installed both of these programs using 'npm install -g ...'

When doing this for real, I think we want to check-in both of these generated files.  Our CI scripts should generate the .d.ts file automatically.  If CI detects that there's a difference in the .d.ts file then that should be a signal that the .js.flow also needs to be updated.  The .js.flow file should not be generated automatically b/c it needs to be tweaked.  Any manual tweaks should be called out in the file using our standard '>KA'/'<KA' markers.

## Test Plan:
- copy contents of .js.flow file into node_modules/@khanacademy/wonder-stuff-core/dist/index.js.flow in wonder-blocks and webapp
- run flow in those repos
- see that there are no errors
